ext/standard/streamsfuncs.c

```diff
@@ -616,93 +616,147 @@ PHP_FUNCTION(stream_get_wrappers)
 }
 /* }}} */
 
-/* {{{ stream_select related functions */
-static int stream_array_to_fd_set(zval *stream_array, fd_set *fds, php_socket_t *max_fd)
-{
-	zval *elem;
-	php_stream *stream;
-	int cnt = 0;
+/* {{{ fd_bigset related macros and typedefs to remove FD_SETSIZE limitation in stream_select */
+typedef struct {
+    char *fds_bits;
+    size_t size;
+} fd_bigset;
+#define FD_BIGSET_ENSURE_CAPACITY(fd, set) \
+    if (((fd) / 8 >= (set)->size)) { \
```

Comment:
``` suggestion
    if (UNEXPECTED(((fd) / 8 >= (set)->size))) { \
```

Reply:
Thanks! Didn't know about that compiler optimization:)